core/src/character.rs

```diff
@@ -1,18 +1,31 @@
+use std::cell::RefCell;
+
 use crate::backend::audio::SoundHandle;
 use crate::binary_data::BinaryData;
 use crate::display_object::{
-    Avm1Button, Avm2Button, Bitmap, EditText, Graphic, MorphShape, MovieClip, Text, Video,
+    Avm1Button, Avm2Button, BitmapClass, EditText, Graphic, MorphShape, MovieClip, Text, Video,
 };
 use crate::font::Font;
-use gc_arena::Collect;
+use gc_arena::{Collect, GcCell};
+use ruffle_render::bitmap::{BitmapHandle, BitmapSize};
+use swf::DefineBitsLossless;
 
 #[derive(Clone, Collect, Debug)]
 #[collect(no_drop)]
 pub enum Character<'gc> {
     EditText(EditText<'gc>),
     Graphic(Graphic<'gc>),
     MovieClip(MovieClip<'gc>),
-    Bitmap(Bitmap<'gc>),
+    Bitmap {
+        #[collect(require_static)]
+        compressed: CompressedBitmap,
+        /// A lazily constructed GPU handle, used when performing fills with this bitmap
+        #[collect(require_static)]
+        handle: RefCell<Option<BitmapHandle>>,
```

Comment:
I feel like this should be an enum. Either we have the handle and we don't care about compressed, or we have compressed and not yet a handle.

Or are there cases where we *do* care about compressed when we've already decompressed it?

Reply:
We could have a bitmap that's only ever used as a gradient fill, which would I ly require us to store the decompressed texture in gpu memory (via the BitmapHandle)

Reply:
Sorry I don't understand (I tried unsuccessfully to decipher the typo - sorry! 😅)

Reply:
Oops, updated the comment

Reply:
Sorry still confused :D

In that case why would you still want the `CompressedBitmap` around?

I'm thinking it's something like:

```rust
enum BitmapType {
    Compressed(CompressedBitmap),
    Decompressed{ handle: RefCell<Option<BitmapHandle>>, avm2_bitmapdata_class: GcCell<'gc, BitmapClass<'gc>> },
}
```

Reply:
How would we handle instantiating the bitmap in your proposal?

Reply:
I'm not sure, but it really feels weird that it'd need the compressed data when we've already uncompressed it...

Not blocker for this PR, but I'd like to come back to this